### PR TITLE
[NFC][SYCL] Split fpga AOC archive tests to address random timeouts

### PR DESCRIPTION
**Pull request by delecui**
Split fpga AOC tests fpga-aoc-archive and fpga-aoc-archive-early to address random timeouts even after the initial split in commit 4af4d4b. This update further divides the tests into smaller units to minimize the number of compile and run sequences.

### CODE REVIEW

**Comment by delecui** [comment]
FYI. Details about the random test timeout can be found in https://jira.devtools.intel.com/browse/CMPLRLLVM-59890

File: sycl/test-e2e/lit.cfg.py [hunk 1]
```diff
@@ -815,6 +815,6 @@ def open_check_file(file_name):
 try:
     import psutil
 
-    lit_config.maxIndividualTestTime = 600
```
Comment (AlexeySachkov) [inline sycl/test-e2e/lit.cfg.py:818]: I'm against this change. Our tests are slow as it is, Allowing them to take 2x time is a bad idea and tests should still be split further if they reach timeout

Comment (wenju-he) [reply sycl/test-e2e/lit.cfg.py:818]: oh, if this PR is impacting all tests, then change in this file is not good

Comment (delecui) [reply sycl/test-e2e/lit.cfg.py]: Got it. I've reverted the timeout change.

File: sycl/test-e2e/AOT/fpga-aoc-archive-early2.cpp [hunk 2]
```diff
@@ -0,0 +1,35 @@
+// Test compiler behaviors for -fintelfpga with -fsycl-link=early.
```
Comment (wenju-he) [inline sycl/test-e2e/AOT/fpga-aoc-archive-early2.cpp:1]: 2 in the filename isn't much meaningful. Rename the file to sycl/test-e2e/AOT/fpga-aoc-archive-early-image.cpp?

Comment (delecui) [reply sycl/test-e2e/AOT/fpga-aoc-archive-early2.cpp:1]: There are three existing test files before the splitting.
| Test file     | Test purpose |
| ------------- | ------------- |
| fpga-aoc-archive.cpp | mixed usage of early and image archives |
| fpga-aoc-archive-early.cpp | usage of early archive  |
| fpga-aoc-archive-image.cpp | usage of image archive|

File names `fpga-aoc-archive2.cpp` and `fpga-aoc-archive-early2.cpp` should indicate the test purpose more clearly than names like `fpga-aoc-archive-early-image.cpp`.

Comment (wenju-he) [reply sycl/test-e2e/AOT/fpga-aoc-archive-early2.cpp:1]: LGTM

**Review by againull — APPROVED** [review]

**Comment by delecui** [comment]
Hi @intel/dpcpp-tools-reviewers, could you help review this pull request? Thanks.

**Review by AlexeySachkov — APPROVED** [review]

**Comment by delecui** [comment]
@intel/llvm-gatekeepers, it looks this PR is ready to merge. Thanks.

**Pull request merged**